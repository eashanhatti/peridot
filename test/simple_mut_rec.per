define x : Record {} = y;
define y : Record {} = x;
