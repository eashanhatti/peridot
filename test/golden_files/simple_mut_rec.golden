((fromList [], fromList [], fromList []),
 Declare
   Obj
   (Rigid
      (RNameIntro
         (UserName "<PLACEHOLDER>") Obj (Id {unId = 0})))
   (RecType (fromList []))
   (Declare
      Obj
      (Rigid
         (RNameIntro
            (UserName "<PLACEHOLDER>") Obj (Id {unId = 1})))
      (RecType (fromList []))
      (Define
         (Rigid
            (RNameIntro
               (UserName "<PLACEHOLDER>") Obj (Id {unId = 0})))
         (GlobalVar
            (Rigid
               (RNameIntro (UserName "y") Obj (Id {unId = 1})))
            False)
         (Define
            (Rigid
               (RNameIntro
                  (UserName "<PLACEHOLDER>") Obj (Id {unId = 1})))
            (GlobalVar
               (Rigid
                  (RNameIntro (UserName "x") Obj (Id {unId = 0})))
               False)
            (Rigid (TypeType Obj))))))
